File: greedy/jump-game-ii/jump-game-ii.ts
function jump(nums: number[]): number {
  const n: number = nums.length;

  let max: number = 0;
  let jumps: number = 0;
  let right: number = 0;

  for (let i = 0; i < n - 1; i++) {
    max = Math.max(max, i + nums[i]);

    if (i === right) {
      jumps++;
      right = max;
    }
  }

  return jumps;
}

File: greedy/jump-game-ii/jump-game-ii.java
class Solution {
  public int jump(int[] nums) {
    int n = nums.length;

    int max = 0;
    int right = 0;
    int jumps = 0;

    for (int i = 0; i < n - 1; i++) {
      max = Math.max(max, i + nums[i]);

      if (i == right) {
        right = max;
        jumps++;
      }

    }

    return jumps;

  }
}
